### PR TITLE
Homepage input validation

### PR DESCRIPTION
**Pull request by tahiya-mongo**
This PR adds type attribute and `maxLength` to validate user inputs on the homepage share ideas/projects form.

### CODE REVIEW

File: src/components/dev-hub/project-sign-up-form.js [hunk 1]
```diff
@@ -73,18 +73,22 @@ const Form = React.memo(({ setSuccess, success }) => {
             </ErrorMessage>
             <StyledInput
                 value={name}
+                maxLength="50"
                 required
                 placeholder="Name"
+                pattern="^[A-Za-zÀ-ÿ ,.'-]+$"
```
Comment (jestapinski) [inline src/components/dev-hub/project-sign-up-form.js:79]: This is great! The only issue is with UX for the error message. [Check out this article](https://webdesign.tutsplus.com/tutorials/html5-form-validation-with-the-pattern-attribute--cms-25145), and can we add a more user-friendly error message?

File: src/components/dev-hub/project-sign-up-form.js [hunk 2]
```diff
@@ -73,18 +73,29 @@ const Form = React.memo(({ setSuccess, success }) => {
...
             <StyledInput
+                type="email"
                 value={email}
                 required
                 placeholder="Email Address"
                 onChange={e => setEmail(e.target.value)}
             />
             <TextArea
                 value={projectDescription}
+                maxLength="250"
```
Comment (greggb) [inline src/components/dev-hub/project-sign-up-form.js:98]: Maybe we can update the placeholder to show this limit?

File: src/components/dev-hub/project-sign-up-form.js [hunk 3]
```diff
@@ -73,20 +73,37 @@ const Form = React.memo(({ setSuccess, success }) => {
...
+                        'Names should only contain letters. e.g. John Doe'
+                    )
+                }
             />
             <StyledInput
+                type="email"
                 value={email}
                 required
                 placeholder="Email Address"
+                pattern="[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,3}$"
```
Comment (jestapinski) [inline src/components/dev-hub/project-sign-up-form.js:93]: the `email` type also gives us some validation checks for free, so I don't think we need this pattern or `onInput` or `onInvalid` here. Were you seeing something else?

Comment (tahiya-mongo) [reply src/components/dev-hub/project-sign-up-form.js:93]: The `email` type validation does not seem to check for the `.com` part. For example, `example@email` is considered valid.

Comment (jestapinski) [reply src/components/dev-hub/project-sign-up-form.js:93]: aha! Nice find!

**Pull request merged**